BitcoinApp/CoinViewController/CoinsViewController.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  CoinsViewController.swift
+//  BitcoinApp
+//
+//  Created by Zhassulan Aimukhambetov on 12/5/19.
+//  Copyright © 2019 Zhassulan Aimukhambetov. All rights reserved.
+//
+
+import UIKit
+import Charts
+
+class CoinsViewController: UIViewController {
+    
+    @IBOutlet weak var bitcoinChartView: LineChartView!
+    @IBOutlet weak var rateValueLabel: UILabel!
+    @IBOutlet weak var periodSegmentControl: UISegmentedControl!
+    @IBOutlet weak var currencySegmentControl: UISegmentedControl!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        guard let currency = Currency(rawValue: currencySegmentControl.selectedSegmentIndex) else { return }
+        guard  let period = Period(rawValue: periodSegmentControl.selectedSegmentIndex) else { return }
+        
+        configureChartView()
+        configureSegmentControl()
+        updateCurrencyValue(currency: currency)
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    //MARK: - @IBActions
+    
+    @IBAction func currencySegmentControlSelect(_ sender: UISegmentedControl) {
+        guard let currency = Currency(rawValue: sender.selectedSegmentIndex) else { return }
+        updateCurrencyValue(currency: currency)
+        guard let period = Period(rawValue: periodSegmentControl.selectedSegmentIndex) else { return }
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    @IBAction func periotSegmentControlSelect(_ sender: UISegmentedControl) {
+        guard let period = Period(rawValue: sender.selectedSegmentIndex) else { return }
+        guard let currency = Currency(rawValue: currencySegmentControl.selectedSegmentIndex) else { return }
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    //MARK: - getData
+    
+    func updateCurrencyHistory(currency: Currency, period: Period) {
+        HistoryNetworkService.getHistory(currency: currency, period: period) { (dataY) in
+            DispatchQueue.main.async {
```

Comment:
Your services should guarantee callback on main queue in another queue that you should pass in the arguments.
It would be easier to make the main queue by default.